contentcuration/contentcuration/frontend/shared/vuex/session/index.js

```diff
@@ -85,20 +100,28 @@ export default {
     },
   },
   actions: {
+    async saveSession(context, currentUser) {
+      // It shouldn't happen but let's make sure that there is
+      // no more current users than one in the table (do not remove
+      // the current user being saved if it already exists in the table
+      // though, because dexie-observable delete listener handler would
+      // be run and depending on timing, it might remove the entry after
+      // it's added and the current user would be lost)
+      await Session.indexNotEqual('id', currentUser.id).delete();
```

Comment:
I think I might solve this in a much better way, and that would be to rather always use a constant key for a current user instead of their id in the session table. That way, we would be really sure that there is only one entry, and I would feel confident removing this table reset completely. It seems that I would need to adjust IndexedDBResource `put` to accept an optional key [as Dexie does](https://dexie.org/docs/Table/Table.put()#syntax). I am not yet confident enough to update our wrapper API, so at first, I wanted to ask. Do you foresee any obstacles or does it sound like a good idea?

Reply:
See my suggestion above - by changing the default id field to something other than `id` we can use that and just set it to `CURRENT` or some other such constant, and we only ever put that user into the DB.

`await Session.delete('CURRENT');` would then achieve the same thing here.

Reply:
That might still cause the problem described in the comment above this line.

Reply:
From now on, we'll ensure having exactly one entry using the constant ID though so I think it's safe to remove this completely.